### ISSUE TITLE
Module name not as expected with jspm@0.17.0-beta.18+

### ISSUE THREAD

**Issue by unional** [jspm/jspm-cli#1972]
I'm using `jspm-git@beta` for a private bitbucket server. Bitbucket server is organized as `<project>/<repo>` e.g. `myteam/myModule`.

in `jspm@0.17.0-beta.17`, when I install the module with `jspm i bit:myteam/myModule`, it is installed with `myModule` as the name of the module.

But since `jspm@0.18.0-beta.18` and onwards, it is installed as `myteam/myModule`.

Reference: https://github.com/Orbs/jspm-git/issues/46

**Comment by unional** [comment]
> From https://github.com/Orbs/jspm-git/issues/46#issuecomment-233018431

I can confirm that it is installed as `myModule` for `jspm@0.17.0-beta-17` or prior on new install.

Do you mean it being installed as `myteam/myModule` is expected?
This would make it very cumbersome to use.
For example you have a private github repo: `http://github.mycompany.local`
You set it up through:

```sh
jspm registry create gh jspm-git
> url: http://github.mycompany.local
...
```

Then you use it as:
```sh
jspm install gh:someone/mymodule
```
and it is installed as `someone/mymodule`

**Comment by unional** [comment]
> From https://github.com/Orbs/jspm-git/issues/46#issuecomment-233025421

Yes if I'm not completely wrong then this is the expected behaviour that has been always like that. If you install modules like jspm i bit:myteam/myrepo then you've to reference them as myteam/myrepo. Only if you do a jspm i myrepo=bit:myteam/myrepo then it'll be aliased. Otherwise what would happen if you do a jspm i bit:myteam/myrepo bit:otherteam/myrepo...? Only one of them can get the myrepo alias.

Anyhow this is a behaviour that can't be influenced by jspm-git as this logic is implemented directly in the jspm core. So you'd have to file an issue there or seek for help in the jspm slack channel.

**Comment by adamburgess** [comment]
Has nothing to do with jspm-git at all - this happens on jspm-github as well. Looks like the name parsing got changed somewhere along the line.

**Comment by unional** [comment]
I can see that it is for `@scope` for npm:
https://github.com/jspm/jspm-cli/releases/tag/0.17.0-beta.18

But this behavior should not apply to `github:myteam/mymodule` or `bitbucket:myteam/mymodule` as there is no `@scope`. A bug then?

**Comment by unional** [comment]
I can see in @guybedford commit description : "fix npm scoped installs, revert github:a/b installs to a/b name by default". But I don't understand the logic why it would affect `github` (and `bitbucket`).

**Comment by guybedford** [comment]
It could be worth considering an option here in the registry API itself for registries to choose themselves how `jspm install registry:x/y` should be aliased.

The rule we used to have was `y` became the alias. But because npm scope installs expect a convention of being required by the full scope name, we changed the convention to `x/y`.

Adding an option to control this alias system could be possible - something like `RegistryEndpoint.prototype.alias = function(fullName) { return fullName.split('/').pop(); }` or even just `RegistryEndpoint.prototype.useShortAlias = true` where the default is false.
